sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-jedis/src/main/java/com/azure/messaging/eventhubs/checkpointstore/jedis/JedisRedisCheckpointStore.java

```diff
@@ -0,0 +1,152 @@
+     * @return Null
+     */
+    @Override
+    public Flux<Checkpoint> listCheckpoints(String fullyQualifiedNamespace, String eventHubName, String consumerGroup) {
+        String prefix = prefixBuilder(fullyQualifiedNamespace, eventHubName, consumerGroup);
+        try (Jedis jedis = jedisPool.getResource()) {
+            ArrayList<Checkpoint> listStoredCheckpoints = new ArrayList<>();
+            Set<String> members = jedis.smembers(prefix);
+            if (members.isEmpty()) {
+                jedisPool.returnResource(jedis);
+                return Flux.fromIterable(listStoredCheckpoints);
+            }
+            for (String member : members) {
+                //get the associated JSON representation for each for the members
+                List<String> checkpointJsonList = jedis.hmget(member, CHECKPOINT);
+                jedisPool.returnResource(jedis);
+                if (checkpointJsonList == null) {
+                    return Flux.error(new IllegalStateException("No checkpoints persist in Redis for the given parameters."));
+                }
+                else {
+                    String checkpointJson = checkpointJsonList.get(0);
+                    Checkpoint checkpoint = DEFAULT_SERIALIZER.deserializeFromBytes(checkpointJson.getBytes(StandardCharsets.UTF_8), TypeReference.createInstance(Checkpoint.class));
+                    listStoredCheckpoints.add(checkpoint);
+                }
+            }
+            return Flux.fromIterable(listStoredCheckpoints);
+        }
+    }
+
+    /**
+     * @param fullyQualifiedNamespace The fully qualified namespace of the current instance of Event Hub
+     * @param eventHubName The Event Hub name from which checkpoint information is acquired
+     * @param consumerGroup The consumer group name associated with the checkpoint
+     * @return Null
+     */
+    @Override
+    public Flux<PartitionOwnership> listOwnership(String fullyQualifiedNamespace, String eventHubName, String consumerGroup) {
+        String prefix = prefixBuilder(fullyQualifiedNamespace, eventHubName, consumerGroup);
+        try (Jedis jedis = jedisPool.getResource()) {
+            Set<String> members = jedis.smembers(prefix);
+            ArrayList<PartitionOwnership> listStoredOwnerships = new ArrayList<>();
+            if (members.isEmpty()) {
+                jedisPool.returnResource(jedis);
+                return Flux.fromIterable(listStoredOwnerships);
+            }
+            for (String member : members) {
+                //get the associated JSON representation for each for the members
+                List<String> partitionOwnershipJsonList = jedis.hmget(member, PARTITION_OWNERSHIP);
+                jedisPool.returnResource(jedis);
+                if (partitionOwnershipJsonList == null) {
+                    return Flux.error(new IllegalStateException("No ownership record persist in Redis for the given parameters."));
```

Comment:
If there's no partition ownership record for this value.. can we just set a default one or keep going and skip this in the returned results? (ie. no one has owned this before.)  I don't know if this is a failed state that we _can't_ recover from.

Reply:
That makes sense! I think skipping this result instead would be a better idea.